back/src/main/java/com/ecoassitant/back/service/impl/CalculServiceImpl.java

```diff
@@ -66,21 +68,33 @@ private ResultatDto resultatForProject(Integer idProject){
         var reponseDonnee = reponseDonneeRepository.findByReponseDonneeKey_Projet(projet.get());
         var calculs = calculRepository.findAll();
 
-        var map = new HashMap<Integer, List<CalculEntity>>();
+        var map = new HashMap<Integer, Map<Integer, List<CalculEntity>>>();
         calculs.forEach(calculEntity -> {
             if (!map.containsKey(calculEntity.getNbCalcul()))
-                map.put(calculEntity.getNbCalcul(), new ArrayList<>());
-            var list = map.get(calculEntity.getNbCalcul());
+                map.put(calculEntity.getNbCalcul(), new HashMap<>());
+            var priorite = map.get(calculEntity.getNbCalcul());
+            if (!priorite.containsKey(calculEntity.getPriorite()))
+                priorite.put(calculEntity.getPriorite(), new ArrayList<>());
+            var list = priorite.get(calculEntity.getPriorite());
             list.add(calculEntity);
-            map.put(calculEntity.getNbCalcul(), list);
+            priorite.put(calculEntity.getPriorite(), list);
+            map.put(calculEntity.getNbCalcul(), priorite);
         });
 
-        map.forEach((k, calcul)->{
-            var calculEntier = new CalculEntier(calcul,reponseDonnee);
-            var executer = calculEntier.execute();
+        map.forEach((k, calculsPriorite)->{
```

Comment:
## Les atomic en java --> à exclure avec les lambdas
C'est dans un contexte de synchronization que tu veux les utiliser, pour faire des opérations atomique sur tes valeurs, mais si tu les stockes pour les utiliser dans une lambda, si ton code est utiliser par plusieurs thread, ça fera tout l'inverse. Les lambda imposent les variables finales pour éviter des comportement imprévisibles si les variables venaient à être modifiés en dehors du score de la lambda (et avec les atomics, ça outrepasse cela)
même si IJ le propose c'est pas bien
De plus, les Atomic forcent une écriture de tes valeurs dans la mémoire (volatile) et avec des opérations atomique, ça rend le code plus lent (peut être de pas beaucoup, mais si on les cumule, ça fini que si)

Globalement, pour ne pas avoir ce genre de problème, mieux vaut passer à des boucles for classique et dommage pour les lambdas, même si c'est stylé

je propose un code équivalent avec des boucles for

```java
        map.forEach((k, calculsPriorite) -> {
            Optional<Double> executer = Optional.empty();
            Optional<Phase> phase = Optional.empty();
            for (var calcul : calculsPriorite.values()) {
                var calculEntier = new CalculEntier(calcul, reponseDonnee);
                executer = calculEntier.execute();
                if (executer.isPresent()) {
                    phase = Optional.ofNullable(calculEntier.getPhase());
                    break;
                }
            }
            
            var intitule = "test" + k;
            if (executer.isPresent() && phase.isPresent()) {
                var aDouble = executer.get();
                switch (phase.get()) {
                    case PLANIFICATION -> resultat.addPlanification(new CalculDto(intitule, aDouble));
                    case DEVELOPPEMENT -> resultat.addDeveloppement(new CalculDto(intitule, aDouble));
                    case DEPLOIEMENT -> resultat.addDeploiement(new CalculDto(intitule, aDouble));
                    case TEST -> resultat.addTest(new CalculDto(intitule, aDouble));
                    case MAINTENANCE -> resultat.addMaintenance(new CalculDto(intitule, aDouble));
                    case HORS_PHASE -> resultat.addHorsPhase(new CalculDto(intitule, aDouble));
                }
            }
        });
```

Reply:
changement a été fait